api/authenticator.py

```diff
@@ -917,10 +917,13 @@ class AuthenticationProvider(OPDSAuthenticationFlow):
         },
         { "key": LIBRARY_IDENTIFIER_FIELD,
           "label": _("Library Identifier Field"),
+          "type": "select",
+          "options": [
+              {"key": LIBRARY_IDENTIFIER_RESTRICTION_BARCODE, "label": _("Barcode")},
+          ],
```

Comment:
Have you tried this out? I think you also need the empty option in `options`, so it would look more like this:

```
 "options": [
              {"key": "", "label": _("Do not gather a library identifier")},
              {"key": LIBRARY_IDENTIFIER_RESTRICTION_BARCODE, "label": _("Gather library identifier from barcode")},
          ],
```

Reply:
I have tried it out, and it seemed okay to me. Whats the concern here? 

The empty option is essentially the same as selecting [`LIBRARY_IDENTIFIER_RESTRICTION_TYPE_NONE `](https://github.com/NYPL-Simplified/circulation/pull/838/files#diff-44bd848f3bcb2624e25c57623ef74c4fR910), in that it doesn't apply a restriction. It seems strange to me to have 2 ways in the UI to turn off the enforcement.

Reply:
It is strange to have two ways to turn off the enforcement; I'd forgotten about the other way and was focused on the oddity of having a select drop-down with only a single option. So I agree that what you have will work.

But the strange thing is that we have two ways of turning off enforcement in the first place, not that we expose both ways in the UI. 

In fact, we have three ways of turning off enforcement, since you can also set the library identifier restriction to a value that matches everything.

So my concern is less about the code and more about the UI, which as it stands requires the following configuration setup for a standalone library:

* Gather library ID from the barcode. [option]
* Don't enforce any restriction on the library ID. [option]
* Don't match the library ID against any particular string. [empty text input]

But a standalone library doesn't have a "library ID" at all. It's not in the barcode; it's not anywhere. I don't want to have to explain to libraries that they should gather a nonexistent library ID from the barcode and do nothing with it. It would make more sense to an administrator of that library to be able to set up the following UI, even though it's technically redundant:

* Don't gather library ID at all. [option]
* Don't enforce any restriction on the library ID. [option]
* Don't match the library ID against any particular string. [empty text input]
 
Ideally the second two configuration options would only show up once you decided to gather the library ID from somewhere, but failing that I think it's best to be able to make it clear in the first configuration option that there is no library ID.

Reply:
I agree it would be nice to be able to configure fields that showed up conditionally based on your other selections, but we'd have to figure out how to represent that and it would also require more front-end work. It's probably not worth it until we have some more settings like that.

For now I think this PR is good as is but some of the description strings could be improved. The library identifier restriction field and library identifier restriction descriptions could say that they won't be used if the library identifier restriction type is no restriction.

Reply:
I agree that it's good enough to change the description strings.

Reply:
Here is my proposed verbiage, let me know if this will suffice and I will make the change. If not alternate suggestions for the text would be appreciated. 

----

`LIBRARY_IDENTIFIER_RESTRICTION_TYPE`
When multiple libraries share an ILS, a person may be able to authenticate with the ILS but not be considered a patron of *this* library. This setting contains the rule for determining whether an identifier is valid for this specific library. 

If this setting it set to 'No Restriction' then the values for *Library Identifier Field* and *Library Identifier Restriction* will not be used.

`LIBRARY_IDENTIFIER_FIELD`
This is the field on the patron record that the *Library Identifier Restriction Type* is applied to, different patron authentication methods provide different values here. This value is not used if *Library Identifier Restriction Type* is set to 'No restriction'.

`LIBRARY_IDENTIFIER_RESTRICTION`
This is the restriction applied to the *Library Identifier Field* using the method chosen in *Library Identifier Restriction Type*. This value is not used if *Library Identifier Restriction Type* is set to 'No restriction'.

Reply:
Sounds good to me.

Reply:
👍